unstructured/metrics/object_detection.py

```diff
@@ -0,0 +1,677 @@
+        mean_recall_per_class = np.zeros(self.num_cls)
+        mean_f1_per_class = np.zeros(self.num_cls)
+
+        if len(document_matchings):
+            matching_info_tensors = [torch.cat(x, 0) for x in list(zip(*document_matchings))]
+
+            # shape (n_class, nb_iou_thresh)
+            (
+                ap_per_present_classes,
+                precision_per_present_classes,
+                recall_per_present_classes,
+                f1_per_present_classes,
+                present_classes,
+            ) = self._compute_detection_metrics(
+                *matching_info_tensors,
+            )
+
+            # Precision, recall and f1 are computed for IoU threshold range, averaged over classes
+            # results before version 3.0.4 (Dec 11 2022) were computed only for smallest value
+            # (i.e IoU 0.5 if metric is @0.5:0.95)
+            mean_precision, mean_recall, mean_f1 = (
+                precision_per_present_classes.mean(),
+                recall_per_present_classes.mean(),
+                f1_per_present_classes.mean(),
+            )
+
+            # MaP is averaged over IoU thresholds and over classes
+            mean_ap = ap_per_present_classes.mean()
+
+            # Fill array of per-class AP scores with values for classes that were present in the
+            # dataset
+            ap_per_class = ap_per_present_classes.mean(1)
+            precision_per_class = precision_per_present_classes.mean(1)
+            recall_per_class = recall_per_present_classes.mean(1)
+            f1_per_class = f1_per_present_classes.mean(1)
+            for i, class_index in enumerate(present_classes):
+                mean_ap_per_class[class_index] = float(ap_per_class[i])
+
+                mean_precision_per_class[class_index] = float(precision_per_class[i])
+                mean_recall_per_class[class_index] = float(recall_per_class[i])
+                mean_f1_per_class[class_index] = float(f1_per_class[i])
+
+        od_evaluation = ObjectDetectionEvaluation(
+            f1_score=float(mean_f1),
+            precision=float(mean_precision),
+            recall=float(mean_recall),
+            m_ap=float(mean_ap),
+        )
+
+        return od_evaluation
+
+
+if __name__ == "__main__":
+    # Example usage
+    prediction_file_paths = [Path("pths/to/predictions.json"), Path("pths/to/predictions2.json")]
+    ground_truth_file_paths = [
+        Path("pths/to/ground_truth.json"),
+        Path("pths/to/ground_truth2.json"),
+    ]
+
+    for prediction_file_path, ground_truth_file_path in zip(
+        prediction_file_paths, ground_truth_file_paths
+    ):
+        eval_processor = ObjectDetectionEvalProcessor.from_json_files(
+            prediction_file_path, ground_truth_file_path
+        )
+
+        metrics: ObjectDetectionEvaluation = eval_processor.get_metrics()
+        print(f"Metrics for {ground_truth_file_path.name}:")
+        from dataclasses import asdict
```

Comment:
Not actually sure how this works in runtime but I think it would be best to either move this import outside of this loop or even to the top of the file

Reply:
I didn't want to import it globally as it's only used by this example in `__main__` but true - I'll move it up outside the loop.